localstack/services/s3/provider.py

```diff
@@ -1150,6 +1161,93 @@ def get_object_attributes(
 
         return response
 
+    def put_bucket_analytics_configuration(
+        self,
+        context: RequestContext,
+        bucket: BucketName,
+        id: AnalyticsId,
+        analytics_configuration: AnalyticsConfiguration,
+        expected_bucket_owner: AccountId = None,
+    ) -> None:
+        moto_backend = get_moto_s3_backend(context)
+        get_bucket_from_moto(moto_backend, bucket)
+
+        store = self.get_store()
+
+        validate_bucket_analytics_configuration(
+            id=id, analytics_configuration=analytics_configuration
+        )
+
+        if bucket not in store.bucket_analytics_configuration.keys():
+            store.bucket_analytics_configuration[bucket] = {}
+        store.bucket_analytics_configuration[bucket][id] = analytics_configuration
+
+    def get_bucket_analytics_configuration(
+        self,
+        context: RequestContext,
+        bucket: BucketName,
+        id: AnalyticsId,
+        expected_bucket_owner: AccountId = None,
+    ) -> GetBucketAnalyticsConfigurationOutput:
+        moto_backend = get_moto_s3_backend(context)
+        get_bucket_from_moto(moto_backend, bucket)
+
+        store = self.get_store()
+
+        analytics_configuration: AnalyticsConfiguration = store.bucket_analytics_configuration.get(
+            bucket, {}
+        ).get(id)
+        if not analytics_configuration:
+            raise NoSuchConfiguration("The specified configuration does not exist.")
+        return GetBucketAnalyticsConfigurationOutput(AnalyticsConfiguration=analytics_configuration)
+
+    def list_bucket_analytics_configurations(
+        self,
+        context: RequestContext,
+        bucket: BucketName,
+        continuation_token: Token = None,
+        expected_bucket_owner: AccountId = None,
+    ) -> ListBucketAnalyticsConfigurationsOutput:
+        moto_backend = get_moto_s3_backend(context)
+        get_bucket_from_moto(moto_backend, bucket)
+
+        store = self.get_store()
+        analytics_configurations: Dict[
+            AnalyticsId, AnalyticsConfiguration
+        ] = store.bucket_analytics_configuration.get(bucket, {})
+        analytics_configurations = dict(sorted(analytics_configurations.items()))
+        analytics_configurations: AnalyticsConfigurationList = list(
+            analytics_configurations.values()
+        )
```

Comment:
I'm not sure I understand what the issue is exactly here? we didn't return the exact same order as AWS? How does AWS sort the list?

We could simplify the code a bit by sorting only the resulting list if AWS sorts on the `Id` value:
```suggestion
        analytics_configurations: AnalyticsConfigurationList = sorted(analytics_configurations.values(), key=lambda x: x["Id"])
```

Reply:
I've updated it but didn't got a chance to push before this review.

Also, Now since the list operation returns same ordered dict as AWS, we are snapshot validating it.